tests/components/vulcan/test_config_flow.py

```diff
@@ -0,0 +1,935 @@
+"""Test the Uonet+ Vulcan config flow."""
+import json
+from unittest import mock
+from unittest.mock import patch
+
+from vulcan import Account
+from vulcan.model import Student
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.vulcan import config_flow, const, register
+from homeassistant.components.vulcan.config_flow import (
+    ClientConnectionError,
+    Keystore,
+    VulcanAPIException,
+)
+from homeassistant.const import CONF_PIN, CONF_REGION, CONF_SCAN_INTERVAL, CONF_TOKEN
+
+from tests.common import MockConfigEntry, load_fixture
+
+fake_keystore = Keystore("", "", "", "", "")
+fake_account = Account(
+    login_id=1,
+    user_login="example@example.com",
+    user_name="example@example.com",
+    rest_url="rest_url",
+)
+
+
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
+    flow = config_flow.VulcanFlowHandler()
+    flow.hass = hass
+
+    result = await flow.async_step_user(user_input=None)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "auth"
+
+
+@mock.patch("homeassistant.components.vulcan.config_flow.Vulcan.get_students")
+@mock.patch("homeassistant.components.vulcan.config_flow.Account.register")
+@mock.patch("homeassistant.components.vulcan.config_flow.Keystore.create")
+async def test_config_flow_auth_success(
+    mock_keystore, mock_account, mock_student, hass
+):
+    """Test a successful config flow initialized by the user."""
+    mock_keystore.return_value = fake_keystore
+    mock_account.return_value = fake_account
+    mock_student.return_value = [
+        Student.load(load_fixture("fake_student_1.json", "vulcan"))
+    ]
+    result = await hass.config_entries.flow.async_init(
+        const.DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "auth"
+    assert result["errors"] is None
+    result = await hass.config_entries.flow.async_configure(
```

Comment:
Please patch the integration setup function on all create entry results and reauth successful to avoid setting up the whole integration. See the scaffold script test template for example.